### PR TITLE
add recommended aggregated api server options

### PR DESCRIPTION
**Pull request by deads2k**
This builds on https://github.com/kubernetes/kubernetes/pull/40947.  It uses the `ApplyTo` pattern to group several related options together.

@sttts

### CODE REVIEW

File: cmd/kube-apiserver/app/server.go [hunk 1]
```diff
@@ -106,14 +106,15 @@ func Run(s *options.ServerRunOptions) error {
 
 	// create config from options
 	genericConfig := genericapiserver.NewConfig().
-		WithSerializer(api.Codecs).
-		ApplyOptions(s.GenericServerRunOptions).
-		ApplyInsecureServingOptions(s.InsecureServing)
+		WithSerializer(api.Codecs)
 
-	if _, err := genericConfig.ApplySecureServingOptions(s.SecureServing); err != nil {
+	s.GenericServerRunOptions.ApplyTo(genericConfig)
```
Comment (sttts) [inline cmd/kube-apiserver/app/server.go]: error handling is missing

File: staging/src/k8s.io/apiserver/pkg/server/options/recommended.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+import (
+	"github.com/spf13/pflag"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/server"
+)
+
+// APIServerOptions contains the recommended options for running an API server
+/// If you add something to this list, it should be in a logical grouping
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/server/options/recommended.go]: nit: / too much

File: staging/src/k8s.io/apiserver/pkg/server/options/recommended.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+type APIServerOptions struct {
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/server/options/recommended.go]: Why not call them CommonOptions or RecommendedOptions? The APIServer is implicit from the package.

File: staging/src/k8s.io/apiserver/pkg/server/options/recommended.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+func (o *APIServerOptions) AddFlags(fs *pflag.FlagSet) {
+	o.Etcd.AddFlags(fs)
+	o.SecureServing.AddFlags(fs)
+	o.Authentication.AddFlags(fs)
+	o.Authorization.AddFlags(fs)
+}
+
+func (o *APIServerOptions) ApplyTo(config *server.Config) error {
+	if err := o.SecureServing.ApplyTo(config); err != nil {
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/server/options/recommended.go]: Etcd has no ApplyTo?

File: cmd/kube-apiserver/app/server.go (hunk 1 above)
Comment (sttts) [inline cmd/kube-apiserver/app/server.go]: wrong patch...

File: staging/src/k8s.io/apiserver/pkg/server/options/recommended.go (hunk 4 above)
Comment (deads2k) [inline staging/src/k8s.io/apiserver/pkg/server/options/recommended.go]: > Etcd has no ApplyTo?

Weird, right?  At least now we know.  Something has gone wrong.

File: staging/src/k8s.io/apiserver/pkg/server/options/recommended.go (hunk 3 above)
Comment (deads2k) [inline staging/src/k8s.io/apiserver/pkg/server/options/recommended.go]: I'll update to `RecommendedOptions`

**Pull request merged**